src/libs/dailyBulletin.ts

```diff
@@ -61,50 +64,39 @@ export async function queryLatest() {
 		throw new InternalError('There was a problem getting the most recent email!', e);
 	}
 
-	// Search through the email for any PDF
+	// Get email subject for the date; generate filename from subject
+	const subject = recentMessage.payload?.headers?.filter(h => h.name === 'Subject')[0].value;
+	const filename = generateFilename(
+		subject ?? '',
+		new Date(parseInt(recentMessage.internalDate!, 10))
+	);
```

Comment:
Is it possible that these headers will ever be missing? You might want to do `recentMessage.payload!.headers!.find(h => h.name === 'Subject').value` instead if not.

Reply:
> I'm a little confused by some of your uses of optional chaining...

I'll often use the `??` to allow something to work momentarily knowing that the logic still works.

With that first example with `subject ?? ''`, if the subject is undefined, than it would cause an error when I call `generateFilename`, so I just replace it with a valid string if for some reason `subject` is undefined. `generateFilename` will return `Null` if a valid date is not found, which would happen if the string passed was simply `''`. A couple lines later the returned `filename` is checked whether its null or not, and will return a proper error.

At least to me its a way to reduce the number of null/undefined checks without hugely increasing the complexity of the code or compromising the ability to troubleshoot issues.

Reply:
Every email will have a subject or else my email forward filter wont even work and the account would never see it.